sentry-android-core/src/test/java/io/sentry/android/core/ActivityFramesTrackerTest.kt

```diff
@@ -102,15 +110,166 @@ class ActivityFramesTrackerTest {
         assertEquals(slowFrames!!.value, 5f)
     }
 
+    @Test
+    fun `sets slow and frozen frames even if start was null`() {
+        val sut = fixture.getSut()
+        val arrayAll = SparseIntArray()
+        arrayAll.put(1, 100)
+        arrayAll.put(20, 5)
+        arrayAll.put(705, 6)
+        val array = arrayOf(arrayAll)
+
+        whenever(fixture.aggregator.metrics).thenReturn(null, array)
+
+        sut.addActivity(fixture.activity)
+        sut.setMetrics(fixture.activity, fixture.sentryId)
+
+        val metrics = sut.takeMetrics(fixture.sentryId)
+
+        val totalFrames = metrics!!["frames_total"]
+        assertEquals(totalFrames!!.value, 111f)
+
+        val frozenFrames = metrics["frames_frozen"]
+        assertEquals(frozenFrames!!.value, 6f)
+
+        val slowFrames = metrics["frames_slow"]
+        assertEquals(slowFrames!!.value, 5f)
+    }
+
+    @Test
+    fun `different activities have separate counts - even when called out of order`() {
+        val sut = fixture.getSut()
+        val array0 = SparseIntArray().also {
+            it.put(1, 100)
+            it.put(20, 5)
+            it.put(705, 6)
+        }.let { arrayOf(it) }
+        val array1 = SparseIntArray().also {
+            it.put(1, 110)
+            it.put(20, 6)
+            it.put(705, 7)
```

Comment:
`m`: Testing with boundary values of the equivalence classes `[0-16] [17-700] [701 - infinity]` makes the tests a bit better. This helps to detect off by 1 errors.
```suggestion
            it.put(16, 110)
            it.put(17, 3) // Splitted up 20 
            it.put(700, 3)
            it.put(701, 7)
```

Reply:
Just used the values that were there in old tests; can update